### PR TITLE
Add TypeSystemSwiftTypeRef suport for protocol composition types.

### PR DESCRIPTION
**Pull request by adrian-prantl**
plus assorted other performance improvements that eliminate SwiftASTContext fallback paths.

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Review by augusto2112 — APPROVED** [review]

File: lldb/test/API/lang/swift/clangimporter/protocol_composition/TestSwiftProtocolComposition.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbutil as lldbutil
+import unittest2
+
+class TestSwiftProtocolComposition(TestBase):
+    @skipUnlessDarwin
+    @swiftTest
+    def test(self):
+        """Test that the extra inhabitants are correctly computed for various
```
Comment (augusto2112) [inline lldb/test/API/lang/swift/clangimporter/protocol_composition/TestSwiftProtocolComposition.py]: I think this comment is from a different test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Pull request merged**